modules/selenium/testcontainers/selenium/__init__.py

```diff
@@ -72,3 +78,44 @@ def get_connection_url(self) -> str:
         ip = self.get_container_host_ip()
         port = self.get_exposed_port(self.port)
         return f"http://{ip}:{port}/wd/hub"
+
+    def with_video(self, video_path: Optional[Path] = None) -> Self:
+        video_path = video_path or Path.cwd()
+
+        self.video = SeleniumVideoContainer()
+        video_folder_path = video_path.parent.resolve()
+        self.video.set_videos_host_path(str(video_folder_path))
+
+        if video_path.name:
+            self.video.set_video_name(video_path.name)
+
+        return self
+
+    def start(self) -> 'DockerContainer':
+        if not self.video:
+            super().start()
+            return self
+
+        network = Network().__enter__()
+
+        self.with_kwargs(network=network.name)
+        super().start()
+
+        self.video \
+            .with_kwargs(network=network.name) \
```

Comment:
I encountered an issue and would appreciate it if you could review it: 
using `with_network` (`network.connect`), I aim to block the container's access from the host via a port. 
Providing the network in the container run command should resolve this issue.

Note:
Using with_network, along with adding a delay before connecting to the network and after running the container, resolved the issue.